Inconsistent tests

Lately I've had to restart the tests in Travis multiple times to get them to pass. I think we have a test that acts wierd. Right now this is the problem:

```
=== RUN   TestTopic/baz.bar.foo
2018/08/12 22:08:19 client: publishing 2f5b0a93-04e8-4601-9d3d-7a33fe97b018
2018/08/12 22:08:19 client: did publish message
2018/08/12 22:08:19 client: got delivery
2018/08/12 22:08:19 server: got delivery on queue foo.# correlation id 4779bc5b-332e-4fd4-9e49-6c7b5c8720f7
bindings_test.go:104 foo.bar.baz does not equal baz.bar.foo
2018/08/12 22:08:19 server: stopped waiting for messages on queue 'foo.#'
2018/08/12 22:08:19 server: stopped waiting for messages on queue 'foo.*'
2018/08/12 22:08:19 server: stopped waiting for messages on queue 'baz.*'
2018/08/12 22:08:19 server: listener exiting gracefully
--- FAIL: TestTopic (0.09s)
    --- FAIL: TestTopic/foo.bar (0.03s)
    	bindings_test.go:107: foo.# NOT called on foo.bar
    --- FAIL: TestTopic/foo.bar.baz (0.00s)
    --- FAIL: TestTopic/baz.bar.foo (0.00s)
    	bindings_test.go:102: foo.# WAS called on baz.bar.foo with body foo.bar.baz
```

If I restart the tests it can suddenly pass.

```
=== RUN   TestTimeout
2018/08/12 22:11:14 server: staring listener: amqp://guest:guest@localhost:5672/
2018/08/12 22:11:14 client: did publish message
2018/08/12 22:11:14 server: waiting for messages on queue 'myqueue'
2018/08/12 22:11:14 server: got delivery on queue myqueue correlation id 795c73c0-dada-422b-bf07-918bb93bf584
2018/08/12 22:11:14 server: got delivery on queue myqueue correlation id ed694979-3696-4b07-8312-4a142d064f43
2018/08/12 22:11:14 client: connecting...
2018/08/12 22:11:14 client: starting up
2018/08/12 22:11:14 client: running publisher
2018/08/12 22:11:14 client: publishing 3d6d38aa-0864-4d61-9d64-22672e8c8076
2018/08/12 22:11:14 client: did publish message
2018/08/12 22:11:14 client: waiting for replies
2018/08/12 22:11:14 client: connecting...
2018/08/12 22:11:14 client: starting up
2018/08/12 22:11:14 server: got delivery on queue myqueue correlation id 3d6d38aa-0864-4d61-9d64-22672e8c8076
2018/08/12 22:11:14 client: could not find where to reply. CorrelationId: 3d6d38aa-0864-4d61-9d64-22672e8c8076
2018/08/12 22:11:14 client: running publisher
2018/08/12 22:11:14 client: publishing 06723efe-2cec-4163-8536-e9de66c5df93
2018/08/12 22:11:14 client: did publish message
2018/08/12 22:11:14 client: waiting for replies
2018/08/12 22:11:14 client: connecting...
2018/08/12 22:11:14 client: starting up
2018/08/12 22:11:14 server: got delivery on queue myqueue correlation id 06723efe-2cec-4163-8536-e9de66c5df93
2018/08/12 22:11:14 client: could not find where to reply. CorrelationId: 06723efe-2cec-4163-8536-e9de66c5df93
2018/08/12 22:11:14 client: running publisher
2018/08/12 22:11:14 client: publishing 7587e272-df4d-45cf-be51-820ca0c6193b
2018/08/12 22:11:14 client: did publish message
2018/08/12 22:11:14 client: waiting for replies
2018/08/12 22:11:14 server: got delivery on queue myqueue correlation id 7587e272-df4d-45cf-be51-820ca0c6193b
2018/08/12 22:11:14 server: stopped waiting for messages on queue 'myqueue'
panic: send on closed channel
goroutine 189 [running]:
github.com/bombsimon/amqp-rpc.(*Server).runHandler.func1(0x6a2348, 0x6c2c00, 0xc4201e8840, 0xc4200c5d90, 0xc4201b5aa0, 0xc4200b6800, 0x6c27c0, 0xc4201b5aa0, 0x0, 0xc420201050, ...)
	/home/travis/gopath/src/github.com/bombsimon/amqp-rpc/server.go:308 +0x170
created by github.com/bombsimon/amqp-rpc.(*Server).runHandler
	/home/travis/gopath/src/github.com/bombsimon/amqp-rpc/server.go:301 +0x22f
FAIL	github.com/bombsimon/amqp-rpc	3.127s
```

--- a/server.go
+++ b/server.go
@@ -277,6 +277,12 @@ func (s *Server) runHandler(handler HandlerFunc, deliveries <-chan amqp.Delivery
 	logger.Infof("server: waiting for messages on queue '%s'", queueName)
 
 	for delivery := range deliveries {
+		// Add one delta to the wait group each time a delivery is handled so
+		// we can end by marking it as done. This will ensure that we don't
+		// close the responses channel until the very last go routin handling a
+		// delivery is finished even though we handle them concurrently.
+		wg.Add(1)
+
 		logger.Infof("server: got delivery on queue %v correlation id %v", queueName, delivery.CorrelationId)
 
 		rw := ResponseWriter{
@@ -311,6 +317,9 @@ func (s *Server) runHandler(handler HandlerFunc, deliveries <-chan amqp.Delivery
 				immediate:  rw.immediate,
 				publishing: *rw.publishing,
 			}
+
+			// Mark the specific delivery as finished.
+			wg.Done()
 		}(delivery)
 	}
 

